frontend/src/views/DifferentialExpression/components/Main/connect.ts

```diff
@@ -58,6 +76,130 @@ export const useConnect = () => {
   const { n_cells: nCellsGroup2, isLoading: isLoadingGroup2 } =
     useProcessedQueryGroupFilterDimensions(queryGroup2);
 
+  const router = useRouter();
+  const { data: availableOrganisms, isLoading: isLoadingOrganisms } =
+    useAvailableOrganisms();
+
+  const { data: allFilterOptions, isLoading: isLoadingFilterOptions } =
+    useQueryGroupFilterDimensions(EMPTY_FILTERS);
+
+  // We have a state machine that ensures the below effects are only triggered on page load.
+  // First select the organism and set organismInitializedFromShareURL to true.
+  useEffect(() => {
+    if (!dispatch || isLoadingOrganisms || organismInitializedFromShareURL)
+      return;
+    const { search } = window.location;
+    const params = new URLSearchParams(search);
+    const organism: string | null = params.get("organism");
+    if (!organism) return;
+
+    const isOrganismValid = availableOrganisms.some(
+      (org) => org.id === organism
+    );
+    if (isOrganismValid) {
+      dispatch(selectOrganism(organism));
+    }
+    setOrganismInitializedFromShareURL(true);
+  }, [
+    dispatch,
+    isLoadingOrganisms,
+    availableOrganisms,
+    organismInitializedFromShareURL,
+  ]);
+
+  // Now, we parse the query groups from the URL and set queryGroupInitializedFromShareURL to true.
+  // We only want to do this if the organism has been initialized from the URL since the valid filter options
+  // will change based on the organism.
+  useEffect(() => {
+    if (
+      !dispatch ||
+      !organismInitializedFromShareURL ||
+      isLoadingFilterOptions ||
+      queryGroupInitializedFromShareURL
+    )
+      return;
+
+    const { search } = window.location;
+    const params = new URLSearchParams(search);
+
+    const paramsToRemove: string[] = [];
+    const cellTypes = params.get("celltypes")?.split(",");
```

Comment:
suggestion - instead of repeating this and paramsToRemove do something like this:

```
    const getParamsAndRemove = (paramName) => {
      const paramValues = params.get(paramName)?.split(",");
      if (paramValues) {
        paramsToRemove.push(paramName);
      }
      return paramValues;
    };
    
    
  const filterOptions = (paramValues, filterTerms) => {
      return filterTerms.filter((term) => paramValues.includes(term.id));
    };
```

and then do:
```
    const cellTypes = getParamsAndRemove("celltypes");
```
etc... for the rest


and then below do:

```
    if (tissues) {
      const tissuesFiltered = filterOptions(tissues, allFilterOptions.tissue_terms);
      dispatch(selectQueryGroup1Filters("tissues", tissuesFiltered));
    }
```
etc.. for the rest

that'll make it more readable and less repetitive imo